specs/CookieManagement.md

```diff
@@ -0,0 +1,491 @@
+
+  /// SameSite status of the cookie which represents the enforcement mode of the cookie.
+  /// The default is COREWEBVIEW2_COOKIE_SAME_SITE_KIND_LAX.
+  [propget] HRESULT SameSite([out, retval] COREWEBVIEW2_COOKIE_SAME_SITE_KIND* sameSite);
+  /// Set the SameSite property.
+  [propput] HRESULT SameSite([in] COREWEBVIEW2_COOKIE_SAME_SITE_KIND sameSite);
+
+  /// The security level of this cookie. True if the client is only to return
+  /// the cookie in subsequent requests if those requests use HTTPS.
+  /// The default is false.
+  /// Note that cookie that requests COREWEBVIEW2_COOKIE_SAME_SITE_KIND_NONE but
+  /// is not marked Secure will be rejected.
+  [propget] HRESULT IsSecure([out, retval] BOOL* secure);
+  /// Set the Secure property.
+  [propput] HRESULT IsSecure([in] BOOL secure);
+}
+
+/// Creates, adds or updates, gets, or or view the cookies. The changes would
+/// apply to the context of the user profile. That is, other WebViews under the
+/// same user profile could be affected.
+[uuid(588C8A15-A28A-4FFD-926B-5E6EE7449E7C), object, pointer_default(unique)]
+interface ICoreWebView2CookieManager : IUnknown {
+  /// Create a cookie object with a specified name, value, domain, and path.
+  /// One can set other optional properties after cookie creation.
+  /// This only creates a cookie object and it is not added to the cookie
+  /// manager until you call AddOrUpdateCookie.
+  /// See ICoreWebView2Cookie for more details.
+  HRESULT CreateCookieWithDetails(
+    [in] LPCWSTR name,
+    [in] LPCWSTR value,
+    [in] LPCWSTR domain,
+    [in] LPCWSTR path,
+    [out, retval] ICoreWebView2Cookie** cookie);
+
+  
+  /// Creates a cookie whose params matches those of the specified cookie.
+  HRESULT CreateCookie(
+    [in] ICoreWebView2StagingCookie* cookieParam,
+    [out, retval] ICoreWebView2StagingCookie** cookie);
+
+  /// Gets a list of cookies matching the specific URI.
+  /// You can modify the cookie objects, call
+  /// ICoreWebView2CookieManager::AddOrUpdateCookie, and the changes
+  /// will be applied to the webview.
+  HRESULT GetCookies(
+    [in] LPCWSTR uri,
+    [in] ICoreWebView2GetCookiesCompletedHandler* handler);
+
+  /// Adds or updates a cookie with the given cookie data; may overwrite
+  /// equivalent cookies if they exist.
+  HRESULT AddOrUpdateCookie([in] ICoreWebView2Cookie* cookie);
+
+  /// Deletes a cookie whose params matches those of the specified cookie.
+  HRESULT DeleteCookie([in] ICoreWebView2StagingCookie* cookie);
+
+  /// Deletes browser cookies with matching name and uri or domain/path pair.
+  /// Cookie name is required.
+  /// If uri is specified, deletes all cookies with the given name where domain
+  /// and path match provided URI.
+  /// If domain is specified, deletes only cookies with the exact domain.
+  /// If path is specified, deletes only cookies with the exact path.
+  HRESULT DeleteCookies([in] LPCWSTR name, [in] LPCWSTR uri, [in] LPCWSTR domain, [in] LPCWSTR path);

```

Comment:
Seems like there's overlap between the uri parameter and the domain and path parameter. A URI has a domain and a path. Seems like it doesn't make sense to specify both a uri parameter and a domain or path parameter?  If that's true, then we should have 2 methods. They could be named something like the following if we think the URI one is more popular?

 * DeleteCookies(name, uri)
 * DeleteCookiesWithDomainAndPath(name, domain, path)